File: src/main/java/no/nav/fo/veilarbregistrering/registrering/resources/StartRegistreringStatusMetrikker.java
package no.nav.fo.veilarbregistrering.registrering.resources;

import no.nav.fo.veilarbregistrering.registrering.bruker.RegistreringType;
import no.nav.metrics.Event;
import no.nav.metrics.MetricsFactory;

class StartRegistreringStatusMetrikker {
    static void rapporterRegistreringsstatus(StartRegistreringStatusDto registreringStatus) {
        Event event = MetricsFactory.createEvent("registrering.bruker.data");
        event.addFieldToReport("erAktivIArena", registreringStatus.getRegistreringType() == RegistreringType.ALLEREDE_REGISTRERT);
        event.addFieldToReport("kreverReaktivering", registreringStatus.getRegistreringType() == RegistreringType.REAKTIVERING);
        event.addFieldToReport("sykmeldUnder39uker", registreringStatus.getRegistreringType() == RegistreringType.SPERRET);
        event.addFieldToReport("sykmeldOver39uker", registreringStatus.getRegistreringType() == RegistreringType.SYKMELDT_REGISTRERING);
        event.addFieldToReport("jobbetSiste6av12Mnd", registreringStatus.getJobbetSeksAvTolvSisteManeder());
        event.report();
    }
}

File: src/main/java/no/nav/fo/veilarbregistrering/registrering/bruker/StartRegistreringStatusMetrikker.kt
package no.nav.fo.veilarbregistrering.registrering.bruker

import no.nav.fo.veilarbregistrering.metrics.Event.Companion.of
import no.nav.fo.veilarbregistrering.metrics.Metric.Companion.of
import no.nav.fo.veilarbregistrering.metrics.InfluxMetricsService
import no.nav.fo.veilarbregistrering.registrering.bruker.resources.StartRegistreringStatusDto

internal object StartRegistreringStatusMetrikker {
    @JvmStatic
    fun rapporterRegistreringsstatus(influxMetricsService: InfluxMetricsService, registreringStatus: StartRegistreringStatusDto) {
        var fields = listOf(
            of("erAktivIArena", registreringStatus.registreringType == RegistreringType.ALLEREDE_REGISTRERT),
            of("kreverReaktivering", registreringStatus.registreringType == RegistreringType.REAKTIVERING),
            of("sykmeldUnder39uker", registreringStatus.registreringType == RegistreringType.SPERRET),
            of("sykmeldOver39uker", registreringStatus.registreringType == RegistreringType.SYKMELDT_REGISTRERING)
        )

        registreringStatus.jobbetSeksAvTolvSisteManeder?.let {
            fields = fields + of("jobbetSiste6av12Mnd", it)
        }

        influxMetricsService.reportFields(
            of("registrering.bruker.data"),
            *fields.toTypedArray()
        )
    }
}